### PR TITLE
Migrate tooling and build process

### PR DESCRIPTION
**Pull request by Xabadu**
Our current tooling and build system has a few vulns that are not super straightforward to fix, due to depending on some unmaintained packages, so to address this, some changes:

### New dev/build workflow
We are migrating to use `parcel` to bundle and build the site moving forward. We'll also now use GitHub actions to build the site automatically upon merge instead of building manually in local. Opted for parcel now to implement something more lightweight and straightforward to set up.

Additionally set up linting and formatting with `eslint` and `prettier`. I also added a pre-commit hook to automatically lint and format code.

### Switching from using mustache into React
This wasn't super needed now, but since there are some ideas we'll be discussing to expand and implement in the future where mustache might fall short, figured it was a good opportunity to make the switch.

### Moved from Stylus to Sass
This was purely due to the Stylus transformer being a bit buggy with parcel, just some minor formatting changes, but works pretty much the same way than before.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,42 +1,43 @@
 {
-  "name": "NodeSchoolSF",
+  "name": "nodeschool-sf",
   "version": "2.0.0",
```
Comment (tlhunter) [inline package.json:3]: 3.0.0?! ;)

**Review by tlhunter — APPROVED** [review]

**Comment by tlhunter** [comment]
good stuff!

File: package.json (hunk 1 above)
Comment (Xabadu) [reply package.json:3]: 😅 yeah, good point. Will update

**Comment by Xabadu** [comment]
Closing in favor of #303

**Pull request closed without merging**